src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapConnection.cs

```diff
@@ -379,7 +378,30 @@ public IAsyncResult BeginSendRequest(DirectoryRequest request, TimeSpan requestT
 
                     s_asyncResultTable.Add(asyncResult, messageID);
 
-                    _fd.BeginInvoke(messageID, operation, ResultAll.LDAP_MSG_ALL, requestTimeout, true, new AsyncCallback(ResponseCallback), requestState);
+                    ResponseCallback(ConstructResponseAsync(messageID, operation, ResultAll.LDAP_MSG_ALL, requestTimeout, true, sync: false), requestState);
+
+                    static async void ResponseCallback(ValueTask<DirectoryResponse> vt, LdapRequestState requestState)
```

Comment:
Probably best to use Task instead of void. If you get a warning at the call site about an unused task, you can suppress it with `_ = `.